c++/include/orc/Vector.hh

```diff
@@ -182,12 +182,20 @@ namespace orc {
     uint64_t getMemoryUsage();
     bool hasVariableLength();
 
+    bool hasElements = false;
+    bool hasPositions = false;
+
     /**
      * The offset of the first element of each list.
      * The length of list i is offsets[i+1] - offsets[i].
      */
     DataBuffer<int64_t> offsets;
 
+    /**
+     * Position of each element in their respective array.
+     */
+    DataBuffer<int64_t> pos;
```

Comment:
`uint32_t` might be enough and `positions` might be better. BTW, I think clients can calculate this in realtime based on `offsets`. Maybe we don't need this field.

Reply:
Agree, ListColumnReader can just populate the offsets and leave the pos calculation done by the client. Will remove this field.